Physical_Engine/Query/MaterialComposition.cs

```diff
@@ -68,6 +69,21 @@ public static MaterialComposition MaterialComposition(this ISurface surface)
             return surface.Construction.IMaterialComposition();
         }
 
+        /***************************************************/
+
+        [Description("Gets all the Materials a BulkMaterial is composed of and in which ratios")]
+        [Input("bulkMaterial", "The BulkMaterial to get the MaterialComposition from")]
+        [Output("materialComposition", "The kind of matter the BulkMaterial is composed of and in which ratios", typeof(Ratio))]
+        public static MaterialComposition MaterialComposition(this BulkMaterial bulkMaterial)
+        {
+            if (bulkMaterial.MaterialComposition == null)
+            {
+                Engine.Reflection.Compute.RecordError("The BulkMaterial MaterialComposition could not be calculated as no Materials have been assigned.");
+                return null;
+            }
+            return Matter.Create.MaterialComposition(bulkMaterial.MaterialComposition().Materials, bulkMaterial.MaterialComposition().Ratios);
```

Comment:
Can just return the property here? (also think the current implementation will lead to stack-overflow if I am not missing something).

Might want to keep the warning as well, if usefull.

```suggestion
            return bulkMaterial.MaterialComposition;
```

If you want to create a new material composition, should at least change to:

```suggestion
            if (bulkMaterial.MaterialComposition == null)
            {
                Engine.Reflection.Compute.RecordError("The BulkMaterial MaterialComposition could not be calculated as no Materials have been assigned.");
                return null;
            }
            return Matter.Create.MaterialComposition(bulkMaterial.MaterialComposition.Materials, bulkMaterial.MaterialComposition.Ratios);
```

To avoid the stack overflow.

Reply:
Thanks @IsakNaslundBh, indeed the last commit led to stack-overflow. I've changed to simply use the properties to work as expected.